Swift5_2.js

```diff
@@ -0,0 +1,111 @@
+        {this._renderCode()}
+      </Highlight>
+    );
+  }
+
+  _renderCode() {
+    const { regex, flags, delimiter, testString, isSubstituting, substString, isGlobal } = this.props;
+
+    const codeString = new CodeString();
+    const includesXFlag = flags.includes("x");
+
+    // Generate code string
+    codeString.append(`import Foundation`)
+    codeString.append()
+    codeString.append(`let pattern = ${this._escapeSwiftString(regex.toString())}`)
+    codeString.append(`let regex = try! NSRegularExpression(pattern: pattern, options: ` + (includesXFlag ? `.anchorsMatchLines` : `[]`) + `)`)
+    codeString.append(`let testString = ${this._escapeSwiftString(testString)}`)
+    codeString.append(`let stringRange = NSRange(location: 0, length: testString.utf16.count)`)
+
+    if (isSubstituting) {
+      codeString.append(`let substitutionString = ${this._escapeSwiftString(substString)}`)
+      if (isGlobal) {
+        codeString.append(`let result = regex.stringByReplacingMatches(in: testString, range: stringRange, withTemplate: substitutionString)`)
+        codeString.append(`print(result)`)
+      } else {
+        codeString.append(`let lookupRange = (testString as NSString).range(of: pattern, options: .regularExpression, range: stringRange)`)
+        codeString.append(`if lookupRange.intersection(stringRange) != nil {`, 4)
+        codeString.append(`let result = regex.stringByReplacingMatches(in: testString, range: lookupRange, withTemplate: substitutionString)`)
+        codeString.append(`print(result)`, 0)
+        codeString.append(`} else {`, 4)
+        codeString.append(`print("No matches were found.")`, 0)
+        codeString.append(`}`)
+      }
+    } else {
+        if (isGlobal) {
+          codeString.append(`let matches = regex.matches(in: testString, range: stringRange)`)
+          codeString.append(`var result: [[String]] = []`)
+          codeString.append(`for match in matches {`, 4)
+          codeString.append(`var groups: [String] = []`)
+          codeString.append(`for rangeIndex in 1 ..< match.numberOfRanges {`, 8)
+          codeString.append(`groups.append((testString as NSString).substring(with: match.range(at: rangeIndex)))`, 4)
+          codeString.append(`}`)
+          codeString.append(`if !groups.isEmpty {`, 8)
+          codeString.append(`result.append(groups)`, 4)
+          codeString.append(`}`, 0)
+          codeString.append(`}`)
+          codeString.append(`print(result)`)
+        } else {
+          codeString.append(`if let firstMatch = regex.firstMatch(in: testString, range: stringRange) {`, 4)
+          codeString.append(`let result: [String] = (1 ..< firstMatch.numberOfRanges).map { (testString as NSString).substring(with: firstMatch.range(at: $0)) }`)
+          codeString.append(`print(result)`, 0)
+          codeString.append(`} else {`, 4)
+          codeString.append(`print("No matches were found.")`, 0)
+          codeString.append(`}`)
+        }
```

Comment:
Here, if I were to follow the PHP template available on GitHub, as well as the other code generators I saw live on the website, we'd end up dumping every "match", something like this:

```
- <NSExtendedRegularExpressionCheckingResult: 0x600000b38f80>{4, 43}{<NSRegularExpression: 0x600002b1fa20> (http[s]?(\S*))(.*) 0x0} #0
  - super: NSRegularExpressionCheckingResult
    - super: NSTextCheckingResult
      - super: NSObject
- <NSExtendedRegularExpressionCheckingResult: 0x600000b39580>{66, 28}{<NSRegularExpression: 0x600002b1fa20> (http[s]?(\S*))(.*) 0x0} #0
  - super: NSRegularExpressionCheckingResult
    - super: NSTextCheckingResult
      - super: NSObject
- <NSExtendedRegularExpressionCheckingResult: 0x600000b3a280>{99, 18}{<NSRegularExpression: 0x600002b1fa20> (http[s]?(\S*))(.*) 0x0} #0
  - super: NSRegularExpressionCheckingResult
    - super: NSTextCheckingResult
      - super: NSObject
```

I don't think the dump above is particularly useful, and oftentimes users won't know how to interpret the match object (`NSTextCheckingResult`). For this reason, I propose printing an array of matching groups (`[[String]]`). This array is more useful and it's probably closer to what users are lookin for.

If this is **not** desired, the highlighted code can be replaced with the code below, that dumps the `NSTextCheckingResult ` objects:

```suggestion
        if (isGlobal) {
          codeString.append(`let matches = regex.matches(in: testString, range: stringRange)`)
          codeString.append(`for match in matches {`, 4)
          codeString.append(`dump(match)`, 0)
          codeString.append(`}`)
        } else {
          codeString.append(`if let firstMatch = regex.firstMatch(in: testString, range: stringRange) {`, 4)
          codeString.append(`print(match)`, 0)
          codeString.append(`} else {`, 4)
          codeString.append(`print("No matches were found.")`, 0)
          codeString.append(`}`)
        }
```

Reply:
Seems fine as is!